### PR TITLE
Expand Ethernet interface intro

### PR DESCRIPTION
**Pull request by SeppoTakalo**
@AnotherButler Please review.

### CODE REVIEW

File: docs/reference/api/connectivity/networksocket/EthInterface.md [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
+// Open a UDP socket
+UDPSocket socket;
+socket.open(&eth);
+```
 
-1. Instantiate the `EthInterface` class.
-1. Call the `connect` function.
-1. Once you connect the EthInterface, you can use it as a
-target for opening [network sockets](/docs/development/reference/network-socket.html).
+### Configuration
```
Comment (AnotherButler) [inline docs/reference/api/connectivity/networksocket/EthInterface.md]: I think we should move the content from this section into the connectivity configuration section.

File: docs/reference/api/connectivity/networksocket/EthInterface.md [hunk 2]
```diff
@@ -1,19 +1,45 @@
 ## Ethernet
 
-The EthInterface provides a C++ API for connecting to the internet over Ethernet.
+The `EthInterface` provides a C++ API for connecting to the internet over Ethernet. By default, this class does not require any configuration. It is able to pick up the default Ethernet driver for the target and select the correct network stack.
```
Comment (AnotherButler) [inline docs/reference/api/connectivity/networksocket/EthInterface.md]: Is there anything else developers should know about what this API is and what it does? This section seems short.

File: docs/reference/api/connectivity/networksocket/EthInterface.md (hunk 1 above)
Comment (SeppoTakalo) [reply docs/reference/api/connectivity/networksocket/EthInterface.md]: I don't see that it fits there.
That configuration section is about build time configuration.

This however, refers to the network configuration, how to get the IP address. It is just that you either call the `set_network()` or not... There is nothing to configure through JSON settings.

Should I change the tittle?
Maybe `IP address configuration` would be better to reflect this.

File: docs/reference/api/connectivity/networksocket/EthInterface.md (hunk 2 above)
Comment (SeppoTakalo) [reply docs/reference/api/connectivity/networksocket/EthInterface.md]: Its hard to see what should be inserted here.

Networking architecture in general is described in #470 and Ethernet MAC drivers will be explained on their [own page](https://github.com/ARMmbed/mbed-os/blob/feature-emac/features/netsocket/emac-drivers/README.md) (to be moved to Handbook later on).

This whole API usage is concised in the 2 line example below in the Usage section.

```
EthernetInterface eth;
eth.connect();
```

**Comment by SeppoTakalo** [comment]
Actually I'll merge this with #470 easier to work when I don't need to wait for PR to be merged.

**Pull request closed without merging**